### PR TITLE
r2: bucket behind access

### PR DESCRIPTION
**Pull request by elithrar**
cc @dcpena 

This PR adds a guide for putting an R2 bucket behind Cloudflare Access.

### CODE REVIEW

File: content/r2/examples/cloudflare-access.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+## 3. Connect a custom domain
+
+{{<Aside type="warning">}}
+
+You should create an Access application before connecting a custom domain to your bucket, as connecting a custom domain will otherwise make your bucket public by default.
+
+{{</Aside>}}
+
+You will need to [connect a custom domain](/r2/buckets/public-buckets/#connect-a-bucket-to-a-custom-domain) to your bucket in order to configure it as an Access application:
```
Comment (kodster28) [inline content/r2/examples/cloudflare-access.md]: For ease of maintenance, might make sense to convert the r2 custom domain steps into a partial and then move the unique guidance (`use the same domain as the access policy`) either above / below the partial.

Happy to commit this directly to the PR if helpful.

File: content/r2/examples/cloudflare-access.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+5. (Optional) Configure the block page policy. This can be changed later.
+6. Configure the **Identity providers** that will be used to protect this domain using Access.
+7. Click **Next**
+8. Enter a **Policy name** and an **Action**. This should be **Allow**, and will enable the group(s) you select to access objects within the bucket behind this Access application.
+9. To **Assign a group** (or groups) and allow access to your bucket, select one or more groups. If you have not created any groups, you will [need to do this first](/cloudflare-one/identity/users/groups/#access-groups). You should **ensure that this group only contains the users within your organization that need access to this R2 bucket**.
+10. Click **Next** and then **Add an application**.
+
+Review the [Cloudflare Access documentation](/cloudflare-one/applications/configure-apps/self-hosted-apps/) to understand how to configure additional Access application options.
```
Comment (kodster28) [inline content/r2/examples/cloudflare-access.md:48]: Validated these steps work through testing, for both access policy + r2 custom domain linked to that policy.

**Review by kodster28 — APPROVED** [review]
Content looks good to me.

Provided one suggestion about maintenance stuff, but also happy to follow up with that in a future PR.

File: content/r2/examples/cloudflare-access.md (hunk 1 above)
Comment (elithrar) [reply content/r2/examples/cloudflare-access.md]: That'd be great if you could @kodster28!

Comment (kodster28) [reply content/r2/examples/cloudflare-access.md]: Okay, just pushed and should be gtg.

**Pull request merged**